src/renderer/components/ReactFlowBox.tsx

```diff
@@ -192,21 +199,91 @@ export const ReactFlowBox = memo(({ wrapperRef, nodeTypes, edgeTypes }: ReactFlo
                         return EMPTY_ARRAY;
                     }
 
-                    const bezierPathCoordinates = getBezierPathValues({
-                        source: sourceP,
-                        sourcePosition: Position.Right,
-                        target: targetP,
-                        targetPosition: Position.Left,
-                    });
+                    const leftNode = getNode(sourceHandle.nodeId);
+                    const rightNode = getNode(e.target);
 
-                    // Here we use Bezier-js to determine if any of the node's sides intersect with the curve
-                    const curve = new Bezier(bezierPathCoordinates);
-                    if (!nodeBB.intersectsCurve(curve)) {
-                        return EMPTY_ARRAY;
+                    const leftNodeIsBreakpoint = leftNode?.type === 'breakPoint';
```

Comment:
You added roughly 100 LOC to a function already 400 LOC long. Could you move this out into a function at the top of this file?